util/netcdf_generator.py

```diff
@@ -52,6 +52,34 @@ def _create_zip(self):
                 shutil.rmtree(temp_dir)
             return tzf.read()
 
+    def _filter_params(self, ds, params):
+        missing_params = []
+        params_to_filter = []
+        default_params = ['deployment', 'id', 'lat', 'lon', 'quality_flag']
+
+        for param in params:
+            if param not in ds.data_vars:
+                missing_params.append(param)
+            else:
+                params_to_filter.append(param)
+
+        if missing_params:
+            log.warning('one or more selected parameters (%s) not found in the dataset', missing_params)
+        if params_to_filter:
+            log.debug('filtering parameters: %s', params_to_filter)
+        else:
+            log.warning('no parameters to remove')
+            return ds
+
+        params_to_filter.extend(default_params)
+        if self.stream_request.include_provenance:
+            params_to_filter.append('provenance')
+
+        for key in ds.data_vars:
+            if key not in params_to_filter:
+                ds = ds.drop(key)
```

Comment:
so this returns a new instance of ds everytime a key is dropped?  this question is just for my clarification.  the way it's written looks like that's the case, and if so then it's okay.

Reply:
Yes. Sadly the documentation for xarray isn't clear that drop is non-destructive. It returns the modified dataset (found after trying it the other way).